BSP/Simulator/Hardware/CAN.py

```diff
@@ -16,54 +16,59 @@ def read():
     # Creates file if it doesn't exist
     os.makedirs(os.path.dirname(file), exist_ok=True)
     if not os.path.exists(file):
-        with open(file, 'w'):
+        with open(file, "w"):
             pass
     message = []
-    with open(file, 'r') as csvfile:
-        fcntl.flock(csvfile.fileno(), fcntl.LOCK_EX)
-        csvreader = csv.reader(csvfile)
-        for row in csvreader:
-            message.append(row)
-        fcntl.flock(csvfile.fileno(), fcntl.LOCK_UN)
-    message = [0, 0] if message == [] else message[0]
-
-    if not message: # If can is empty list
-        message.extend(["Empty", "Empty"])
-    return message
+    try:
+        with open(file, "r") as csvfile:
+            fcntl.flock(csvfile.fileno(), fcntl.LOCK_EX)
+            csvreader = csv.reader(csvfile)
+            for row in csvreader:
+                message.append(row)
+            fcntl.flock(csvfile.fileno(), fcntl.LOCK_UN)
+        try:
+            return [message[0][0], message[0][1]]
+        except IndexError:
+            return ["", ""]
+    except Exception:
```

Comment:
Do you think it would be maybe better to handle this exception (and the other ones involving IO) as an IOError? Just for style purposes, because I think that's the only type of exception that this could cause.

Reply:
I was mainly just trying to catch anything that could get dropped in here. I think it should be fine this way if you're okay with it.